### PR TITLE
Enable legend filter hier cluster

### PR DESCRIPTION
**Pull request by congyu-lu**
## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by xzhou82** [comment]
with gdc hiercluster breaks with this backend err:

TypeError: Cannot read properties of undefined (reading 'alias')
    at Object.getGeneAlias (/Users/xzhou1/dev/sjpp/gdc/active/dataset/gdc.hg38.js:138:6)
    at null.getSampleData (webpack://sjcrh/proteinpaint/server/src/termdb.matrix.js:116:51)

sorry i thought i had posted this somewhere

**Comment by siosonel** [comment]
Seems to work

Screenshot 2024-01-01 at 7 22 53 PM

**Comment by xzhou82** [comment]
the other gdc hierclutster examples work. can u try the `GDC HGG` link? it breaks

**Comment by siosonel** [comment]
> the other gdc hierclutster examples work. can u try the `GDC HGG` link? it breaks

@xzhou82 Yes, that breaks. Please pull https://github.com/stjude/sjpp/pull/274 from the sjpp repo, that should fix the error

**Comment by xzhou82** [comment]
to replace `passLegendFilterSampleLst[]` using  `filterJoin()` with this.state.config.legendValueFilter

File: client/termdb/TermdbVocab.js [hunk 1]
```diff
@@ -743,7 +743,7 @@ export class TermdbVocab extends Vocab {
 			if (opts.filter0) init.body.filter0 = opts.filter0 // avoid adding "undefined" value
 			if (opts.isHierCluster) init.body.isHierCluster = true // special arg from matrix, just pass along
 
-			if (tw.term.id && currentGeneNames.length) {
+			if (tw.term.id && currentGeneNames.length && !opts.isHierCluster) {
```
Comment (xzhou82) [inline client/termdb/TermdbVocab.js:746]: i've already fixed this in master. revert this change to avoid conflict

**Review by xzhou82 — APPROVED** [review]
